Drop custom json encoding for tool test API endpoint

nvk747 opened the pull request.

nvk747 committed:
make isinstance check on map python2.7 compatible

nsoranzo commented on lib/galaxy/webapps/galaxy/api/tools.py:
```diff
@@ -205,7 +206,7 @@ def test_data(self, trans, id, **kwd):
         def json_encodeify(obj):
             if isinstance(obj, OrderedDict):
                 return dict(obj)
```
This doesn't seem to be needed after d6cf262ffe9c7c0a4eac4e47aa750340c8f0010e

nsoranzo commented on lib/galaxy/webapps/galaxy/api/tools.py:
```diff
@@ -205,7 +206,7 @@ def test_data(self, trans, id, **kwd):
                 return dict(obj)
-            elif isinstance(obj, map):
+            elif inspect.isclass(map) and isinstance(obj, map):
                 return list(obj)
```
@mvdbeek This also shouldn't be needed any more after https://github.com/galaxyproject/galaxy/pull/9416 ?

I think we may just drop the method.

mvdbeek replied:
Yep, done that. Targeting dev for more extended test coverage, we can backport after merge

mvdbeek committed:
Custom json encoding of test_data response is not neccessary anymore

Since we swapped out odict for OrderedDict (and OrderedDict can be
serialized to JSON), and there are no map objects in the tool test
response.
